getParameterProfiles.m

```diff
@@ -69,178 +69,198 @@
-        otherwise
-            error('Unknown profile computationg method');
+    if (~isfield(options.profileOptimizationOptions, 'MaxFunEvals') ...
+                || isempty(options.profileOptimizationOptions.MaxFunEvals)) 
+        options.profileOptimizationOptions.MaxFunEvals = 200 * parameters.number;
     end
-    options.parameter_index = sort(unique([options.profile_optim_index options.profile_integ_index]));
-else
-    switch options.profile_method
-        case 'optimization'
-            options.profile_optim_index = options.parameter_index;
-            if ~isempty(options.profile_integ_index)
-                error('Some profiles seem to be computed twice. Please redefine consistent options!');
-            end
-            
-        case 'integration'
-            options.profile_integ_index = options.parameter_index;
-            if ~isempty(options.profile_optim_index)
-                error('Some profiles seem to be computed twice. Please redefine consistent options!');
-            end
-            
-        case 'mixed'
-            if (length(unique([options.profile_optim_index, options.profile_integ_index])) ~= length([options.profile_optim_index, options.profile_integ_index])) ...
-                    || (length([options.profile_optim_index, options.profile_integ_index]) ~= length(options.parameter_index))
-                error('Inconsistent settings for indices in profile calculation.');
-            end
-            options.parameter_index = sort(unique([options.profile_optim_index options.profile_integ_index]));
-        
-        otherwise
-            error('Unknown profile computationg method');
+    if (isempty(options.MAP_index))
+        options.MAP_index = 1;
     end
-end
 
-%% Initialization and figure generation
-fh = [];
-switch options.mode
-    case 'visual'
-        if isempty(options.fh)
-            fh = figure('Name','getParameterProfiles');
-        else
-            fh = figure(options.fh);
+    % Process, which profiles should be computed in which manner
+    if strcmp(options.profile_method, 'default')
+        if (isempty(options.profile_optim_index) && isempty(options.profile_integ_index))
+            options.profile_method = 'optimization';
+        elseif (~isempty(options.profile_optim_index) && isempty(options.profile_integ_index))
+            options.profile_method = 'optimization';
+        elseif (isempty(options.profile_optim_index) && ~isempty(options.profile_integ_index))
+            options.profile_method = 'integration';
+        elseif (~isempty(options.profile_optim_index) && ~isempty(options.profile_integ_index))
+            options.profile_method = 'mixed';
         end
-    case 'text'
-        fprintf(' \nProfile likelihood caculation:\n===============================\n');
-    case 'silent' % no output
-        % Force fmincon to be silent.
-        options.profileReoptimizationOptions.display = 'off';
-end
+    end    
+    if isempty(options.parameter_index)

```

Comment:
please simplify this `if else` clause by setting `options.parameter_index` to `1:parameters.number` and remove code duplication.